src-test/org/etools/j1939_84/controllers/part2/Part02Step07ControllerTest.java

```diff
@@ -0,0 +1,1063 @@
+
+        OBDModuleInformation obdModule0x02 = createOBDModuleInformation(0x02,
+                                                                        2,
+                                                                        "SuperMan",
+                                                                        "TheCrystalIcePalace",
+                                                                        "ST109823456",
+                                                                        "Air");
+
+        OBDModuleInformation obdModule0x03 = createOBDModuleInformation(0x03,
+                                                                        3,
+                                                                        "WonderWoman",
+                                                                        "TheLair",
+                                                                        "WW109877654",
+                                                                        "Lasso");
+
+        List<OBDModuleInformation> obdModules = new ArrayList<>();
+        obdModules.add(obdMoule0x00);
+        obdModules.add(obdModule0x01);
+        obdModules.add(obdModule0x02);
+        obdModules.add(obdModule0x03);
+
+        when(dataRepository.getObdModules()).thenReturn(obdModules);
+
+        when(vehicleInformationModule.reportComponentIdentification(any()))
+                .thenReturn(new RequestResult<>(false,
+                                                List.of(packet0x00, packet0x01, packet0x02),
+                                                List.of()));
+
+        when(vehicleInformationModule.reportComponentIdentification(any(), eq(0x00)))
+                .thenReturn(new BusResult<>(false, packet0x00));
+        when(vehicleInformationModule.reportComponentIdentification(any(), eq(0x01)))
+                .thenReturn(new BusResult<>(false, packet0x01));
+        when(vehicleInformationModule.reportComponentIdentification(any(), eq(0x02)))
+                .thenReturn(new BusResult<>(false, packet0x02));
+        when(vehicleInformationModule.reportComponentIdentification(any(), eq(0x03)))
+                .thenReturn(new BusResult<>(false, Optional.empty()));
+
+        runTest();
+
+        verify(dataRepository).getObdModules();
+
+        verify(mockListener).addOutcome(PART_NUMBER,
+                                        STEP_NUMBER,
+                                        FAIL,
+                                        "6.2.7.2.a There are no positive responses to a DS Component ID request from Transmission #1 (3)");
+        verify(mockListener).addOutcome(PART_NUMBER,
+                                        STEP_NUMBER,
+                                        WARN,
+                                        "6.2.7.5.a - Transmission #1 (3) did not provide a positive respond to global query while engine running");
+
+        verify(reportFileModule).addOutcome(PART_NUMBER,
```

Comment:
We don't need to be verifying the interaction with the reportFileModule.  The Controller doesn't actually interact with it.